lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -86,29 +115,19 @@ static void postFileOpen(void*)
         if(layer)
         {
           layer->setLayerAndClearAttribute(stage->GetSessionLayer());
-        }
-      }
-    }
-  }
-  {
-    MItDependencyNodes iter(MFn::kPluginDependNode);
-    for(; !iter.isDone(); iter.next())
-    {
-      fn.setObject(iter.item());
-      if(fn.typeId() == nodes::Layer::kTypeId)
-      {
-        // now go and fix up each of the layer nodes in the scene
-        nodes::Layer* layerPtr = (nodes::Layer*)fn.userNode();
-        MPlug plug = layerPtr->nameOnLoadPlug();
-        MString path = plug.asString();
-        if(path.length() && path.substring(0, 3) != "anon")
-        {
-          SdfLayerHandle layer = SdfLayer::FindOrOpen(path.asChar());
-          LAYER_HANDLE_CHECK(layer);
-          layerPtr->setLayerAndClearAttribute(layer);
-        }
-        else
-        {
+          layer->init(proxy, stage->GetSessionLayer());
+
+          // initialise and fixup each child and sublayer
```

Comment:
Stick these sublayer and childlayer loops into a member function into Layer.h/cpp and call if from here